contracts/ERC-721/IERC721PoolFactory.sol

```diff
@@ -41,4 +55,31 @@ interface IERC721PoolFactory {
     ///
     /// @param asset The underlying ERC-721 asset contract address.
     function createPool(address asset) external;
+
+    /// @notice Rescue the last NFT of a pool.
+    ///
+    /// @dev Emits a {RescueLastNFT} event.
+    ///
+    /// @dev Requirements:
+    /// - Can only rescue the last NFT of a pool.
+    /// - Can only be called by the owner.
+    /// - The pool must exist.
+    ///
+    /// @param asset The underlying ERC-721 asset contract address.
+    function rescueLastNFT(address asset, address to) external;
+
+    /// @notice Set the ENS name of a pool.
+    ///
```

Comment:
@surbhiaudichya I think a `SetENSName` event should be emitted in the implementation.

Reply:
Hey @scorpion9979, 
Could you please provide more details on what you mean by "emitting it in the implementation"?
As of now, the `ENSNameSet` event is emitted within the `setENSName` function in the `ERC721Pool` contract, which seems appropriate for capturing the ENS name setting action. I believe this approach provides clarity and transparency.